### PR TITLE
Generalize helpers for indenting and prefixing

### PR DESCRIPTION
**Pull request by cameel**
Another refactor related to #14433.

This one comes from the fact that at one point I needed a function to indent a chunk of output and I found several, neither of them accessible to me.

In the end I changed the way expectations are formatted and did not need that function (which is why #14506 does not depend on this refactor) but since I already cleaned this up, I think we should still merge it.

### CODE REVIEW

File: test/libsolidity/GasTest.cpp [hunk 1]
```diff
@@ -152,11 +152,3 @@ TestCase::TestResult GasTest::run(ostream& _stream, string const& _linePrefix, b
...
 	}
 }
-
-void GasTest::printSource(ostream& _stream, string const& _linePrefix, bool) const
-{
-	string line;
-	istringstream input(m_source);
-	while (getline(input, line))
-		_stream << _linePrefix << line << std::endl;
-}
```
Comment (cameel) [inline test/libsolidity/GasTest.cpp]: This is superfluous. The inherited version from `TestCase` does the same thing.

**Review by matheusaaguiar — DISMISSED** [review]

**Review by matheusaaguiar — COMMENTED** [review]
LGTM

**Review by matheusaaguiar — APPROVED** [review]

File: libsolutil/StringUtils.h [hunk 2]
```diff
@@ -168,12 +168,44 @@ inline bool isDigit(char _c)
...
+	std::string const& _prefix,
+	bool _trimPrefix = true
+);
+
+/// Prints to a stream, adding a prefix to every line in the input.
+/// Assumes \n as the line separator.
+/// @param _trimPrefix If true, the function avoids introducing trailing whitespace on empty lines.
+///     This is achieved by removing trailing spaces from the prefix on such lines.
+///     Note that tabs and newlines are not removed, only spaces are.
+/// @param _finalNewline If true, an extra \n will be printed at the end of @a _input if it does
```
Comment (nikola-matic) [inline libsolutil/StringUtils.h:192]:
```suggestion
/// @param _finalNewline If true, an extra \n will be printed at the end of @_input if it does
```

File: libsolutil/StringUtils.h [hunk 3]
```diff
@@ -168,12 +168,44 @@ inline bool isDigit(char _c)
...
+	bool _ensureFinalNewline = true
+);
+
+/// Adds a standard indent of 4 spaces to every line in the input.
+/// Assumes \n as the line separator.
+/// @param _indentEmptyLines If true, the indent will be applied to empty lines as well, resulting
+///     such lines containing trailing whitespace.
+inline std::string indent(std::string const& _input, bool _indentEmptyLines = false)
+{
+	return prefixLines(_input, "    ", !_indentEmptyLines);
```
Comment (nikola-matic) [inline libsolutil/StringUtils.h:208]:
```suggestion
	return prefixLines(_input, std::string(4, ' '), !_indentEmptyLines);
```

I think it's clearer this way, at least with regards to the number of spaces in the indentation.

File: libsolutil/StringUtils.h (hunk 2 above)
Comment (cameel) [reply libsolutil/StringUtils.h:192]: What? Is this even a correct doxygen syntax? I think that `@` must be followed by a [command](https://www.doxygen.nl/manual/commands.html) name.

Well, not that we're following this strictly, but still, this seems incorrect.

File: libsolutil/StringUtils.h (hunk 3 above)
Comment (cameel) [reply libsolutil/StringUtils.h:208]: I don't mind either way, but the PR is merged now so I'll leave it as is (the comment was posted while the PR was already on automerge and I did not notice it in time).

**Pull request merged**